### PR TITLE
Some refactoring

### PR DESCRIPTION
**Pull request by vhf**
Less code :)

### CODE REVIEW

File: packages/mdast-util-split/src/index.js [hunk 1]
```diff
@@ -58,22 +29,21 @@ class Splitter {
...
+      // we are at the root
       return
     }
-    if ((node.type !== 'heading' || node.depth !== this.depth) &&
-      parent.type === 'root' && this.lastIndex === -1) {
-      this.introduction.children.push(node)
-      return
-    }
+
     if (node.type === 'heading' && node.depth === this.depth) {
```
Comment (vhf) [inline packages/mdast-util-split/src/index.js:37]: l.70 `(node.type === 'heading' && node.depth === this.depth)` was the inverse of l.65 `(node.type !== 'heading' || node.depth !== this.depth)`

File: packages/mdast-util-split/__tests__/index.js [hunk 2]
```diff
@@ -118,98 +118,11 @@ test('default parameter with canonical text', () => {
...
-  })
-})
-
 test('no heading', () => {
-  const noHeadingText = text.replace(/#/g, '')
-  expect(() => doSplit(noHeadingText, {})).toThrowError('No heading')
+  const headingStripped = text.replace(/#/g, '')
+  const result = doSplit(headingStripped, {})
+  expect(result.trees).toHaveLength(0)
+  expect(result.introduction.type).toBe('root')
```
Comment (artragis) [inline packages/mdast-util-split/__tests__/index.js:125]: so you changed the behaviour?

Comment (vhf) [reply packages/mdast-util-split/__tests__/index.js:125]: Yep, I forgot to mention this change. Throwing makes things more complicated for no real benefit.

I'm not sure about your use case for this but people should rightfully expect this package to split at the depth they have specified and get the appropriate result whether the specified depth exists or not. Some people might want to automate this split for exports for instance, the content before the first heading to split is considered the introduction. This should not fail.

Another way to look at it is:

```shell
$ python
>>> 'abc'.split(',')
['abc']
$ node
> 'abc'.split(',')
[ 'abc' ]
```

Comment (artragis) [reply packages/mdast-util-split/__tests__/index.js:125]: true enouh. I merge your PR. Just need to update the doc now :)

**Pull request merged**